src/DotNetty.Buffers/UnpooledHeapByteBuffer.cs

```diff
@@ -217,6 +218,19 @@ public override long GetLong(int index)
             return this._GetLong(index);
         }
 
+        public override int GetMedium(int index)
+        {
+            this.EnsureAccessible();
+            return this._GetMedium(index).ToMediumInt();
```

Comment:
the fact that you need to call `.ToMediumInt()` here raises a question as to what _GetMedium returns then. Would it make sense to pull `.ToMediumInt()` into _GetMedium?